lib/rom/sql/attribute.rb

```diff
@@ -151,7 +152,25 @@ def to_sym
       #
       # @api public
       def is(other)
-        __cmp__(:'=', other)
+        if NONSTANDARD_EQUALITY_VALUES.include?(other)
+          __cmp__(:IS, other)
+        else
+          __cmp__(:'=', other)
+        end
+      end
```

Comment:
Ended up with failing specs because __cmp__ seems to handle passing types too: https://github.com/rom-rb/rom-sql/blob/master/lib/rom/sql/attribute.rb#L261